### PR TITLE
plasmidfinder with control of version in test data

### PR DESCRIPTION
**Pull request by pimarin**
just a proper PR for plasmidfinder

### CODE REVIEW

**Comment by pimarin** [comment]
I failed to add the escape character to the date

**Comment by pimarin** [comment]
I added the control for CI tests to avoid error with the date

**Comment by pimarin** [comment]
- [X] Update in same time as the related tool
- [X] Planemo pass test
- [X] Test using galaxy and testoolshed with the tool

**Comment by pimarin** [comment]
Is it okay ?

**Comment by bgruening** [comment]
The tests are failing after your latest commits unfortunately :(

**Comment by pimarin** [comment]
I failed to pass the problem of the date, when using `\d\d\d\d-\d\d-\d\d`, it didn't pass when I have difference only on date

**Comment by pimarin** [comment]
@bgruening  any idea to fixe the problem with the `\d` which always failed in tests ?

File: data_managers/data_manager_fetch_plasmidfinder/data_manager_conf.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<data_managers>
+    <data_manager tool_file="data_manager/plasmidfinder_fetch_database.xml" id="plasmidfinder_fetch_database" version="@TOOL_VERSION@">
+        <data_table name="plasmidfinder_database">
+            <output>
+                <column name="value" />
+                <column name="name" />
+                <column name="date" />
```
Comment (bernt-matthias) [inline data_managers/data_manager_fetch_plasmidfinder/data_manager_conf.xml:8]: Sorry for the delay. Can you remind me how the stored dates in this column is formatted and used in the tool.

**Comment by bernt-matthias** [comment]
Yes. That's one question. I will have a look. 

My question was more how the data table is used in the plasmid finder tool. In particular how the date column is used there.

**Comment by pimarin** [comment]
Humm, the tool use name for user selection into galaxy and value to make the path to galaxy manager, date is just added in the .loc file but not used (value, name, date and path)

```
plasmidfinder_1307168b1ce7_2023-02-21	2.1_2023-02-21	2023-02-21	plasmidfinder-db
```

**Comment by pimarin** [comment]
@bernt-matthias  I changed the datamanager to integrate the date of the database to be informative to the user and test are based on dumped version but not the "latest" option. "latest" downlaod the last commit so it's impossible to fix the date

File: data_managers/data_manager_fetch_plasmidfinder/data_manager/macro.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <token name="@GIT_VERSION@">2.34.1</token>
+    <token name="@GIT_DB@">4.0.10</token>
+    <token name="@PYTHON_VERSION@">3.11.3</token>
+    <token name="@VERSION_SUFFIX@">0</token>
+    <token name="@PROFILE@">21.05</token>
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="@PYTHON_VERSION@">python</requirement>
+            <requirement type="package" version="@GIT_PYTHON_VERSION@">gitpython</requirement>
+            <requirement type="package" version="@GIT_VERSION@">git</requirement>
```
Comment (bernt-matthias) [inline data_managers/data_manager_fetch_plasmidfinder/data_manager/macro.xml:13]: Is this extra `git` requirement needed?

File: data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+
```
Comment (bernt-matthias) [inline data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.xml:1]:
```suggestion
```

File: data_managers/data_manager_fetch_plasmidfinder/data_manager_conf.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<data_managers>
+    <data_manager tool_file="data_manager/plasmidfinder_fetch_database.xml" id="plasmidfinder_fetch_database" version="@TOOL_VERSION@">
```
Comment (bernt-matthias) [inline data_managers/data_manager_fetch_plasmidfinder/data_manager_conf.xml:3]:
```suggestion
    <data_manager tool_file="data_manager/plasmidfinder_fetch_database.xml" id="plasmidfinder_fetch_database">
```

FYI https://github.com/galaxyproject/galaxy/pull/12255

**Review by bernt-matthias — COMMENTED** [review]
Does the tool need any updates?

**Comment by pimarin** [comment]
> Does the tool need any updates?

The tool don't have any update, only the DM wasn't validated since I wrapped the tool.

**Review by bernt-matthias — APPROVED** [review]
Looks good to me. Also happy @bgruening ?

File: data_managers/data_manager_fetch_plasmidfinder/.shed.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+categories:
+- Data Managers
+description: Install PlasmidFinder databases
+long_description: |
+  This data manager update to the latest databases for plasmidfinder tool
+name: data_manager_fetch_plasmidfinder
+owner: iuc
+homepage_url: https://github.com/galaxyproject/tools-iuc/
```
Comment (bernt-matthias) [inline data_managers/data_manager_fetch_plasmidfinder/.shed.yml:8]: Maybe this URL should link to the plasmidfinder Website?

**Review by bernt-matthias — APPROVED** [review]
One last comment from my side. Thanks for all the work.

**Pull request merged**